### PR TITLE
feat: admin user create for e2e tests

### PR DESCRIPTION
**Pull request by sivanpayyadakath**
Create an admin user and share it’s access token for testing admin routes.

### CODE REVIEW

File: test/test-utils.ts [hunk 1]
```diff
@@ -39,6 +46,40 @@ export const createDBEntities = async (): Promise<void> => {
...
 };
 
+export const createAdminUser = async (
+  app: INestApplication,
+): Promise<AuthTokenOutput> => {
+  const defaultAdmin: CreateUserInput = {
+    name: 'Default Admin User',
+    username: 'default-admin',
+    password: 'default-admin-password',
+    roles: [ROLE.ADMIN],
```
Comment (yashmurty) [inline test/test-utils.ts:56]: @sivanpayyadakath 
I don't think we should allow any user to register themselves as an `ADMIN`. :thinking: 
Let's discuss how we can solve this issue.

File: test/user/user.e2e-spec.ts [hunk 2]
```diff
@@ -30,39 +30,14 @@ describe('UserController (e2e)', () => {
...
-      .send(registerInput)
-      .expect(HttpStatus.CREATED);
-
-    const tokenResponse = await request(app.getHttpServer())
-      .post('/auth/login')
-      .send(loginInput)
-      .expect(HttpStatus.OK);
-
-    accessToken = tokenResponse.body.data.accessToken;
+    authTokenForAdmin = await createAdminUser(app);
```
Comment (yashmurty) [inline test/user/user.e2e-spec.ts:33]: really nice refactor. All this heavy work is now done in `test-util`. :100:

**Review by yashmurty — CHANGES_REQUESTED** [review]
Have a concern regarding how admin user is being created. 
Let's discuss this.

File: test/user/user.e2e-spec.ts (hunk 2 above)
Comment (yashmurty) [inline test/user/user.e2e-spec.ts:33]: @sivanpayyadakath 
How about the function returning `{ adminUser, authTokenForAdmin }`

Do you think the returned `adminUser` would be useful?

Comment (sivanpayyadakath) [reply test/user/user.e2e-spec.ts:33]: @yashmurty 
Yeah. I would be a nice idea. 👍

File: test/auth/auth.e2e-spec.ts [hunk 3]
```diff
@@ -29,6 +31,15 @@ describe('AuthController (e2e)', () => {
     app = moduleRef.createNestApplication();
     app.useGlobalPipes(new ValidationPipe());
     await app.init();
+
+    ({ authTokenForAdmin } = await seedAdminUser(app));
```
Comment (yashmurty) [inline test/auth/auth.e2e-spec.ts:35]: :eyes:

**Review by yashmurty — APPROVED** [review]
LGTM! :tada:

**Pull request merged**